xmodule/services.py

```diff
@@ -308,3 +312,81 @@ def _handle_deprecated_progress_event(self, block, event):
         # in order to avoid duplicate work and possibly conflicting semantics.
         if not getattr(block, 'has_custom_completion', False):
             self.completion_service.submit_completion(block.scope_ids.usage_id, 1.0)
+
+
+class ShowAnswerService(Service):
+    """
+    An XBlock Service that allows XModules to define if the answers are visible.
+    """
+    @classmethod
+    def answer_available(
+            cls, show_answer='', show_correctness='', past_due=False, attempts=0,
+            is_attempted=False, is_correct=False, required_attempts=0,
+            max_attempts=0, used_all_attempts=False, closed=False,
+            has_staff_access=False):
+        """
+        Returns whether correctness is available now, for the given attributes.
+        """
+        if not show_correctness:
+            # If correctness is being withheld, then don't show answers either.
+            return False
```

Comment:
I believe that `ShowAnswerService` and  `ShowCorrectnessService` services are too specific. How about we combine them into a single service called `ResultService` (or `SolutionService`, `AnswerService`, `ProblemService`, etc.)?

Reply:
Now it is called `ResultService` 😉